api-list.md

```diff
@@ -43,100 +61,117 @@ API Root Path
+`POST /organizations/:org_id/containers` - create a new container
 
-`DELETE organizations/:org_id/containers/:id?:id2` - remove node `id2` from container with `id`
+`PATCH /organizations/:org_id/containers/:id/children` - **ADD** nodes `childNodeId1` and `childNodeId2` to container with `id`
+```
+[ "childNodeId1", "childNodeId2", ... ]
+```
 
-### Surveys
+`PUT organizations/:org_id/containers/:id/children` - **SET** nodes `childNodeId1` and `childNodeId2` to container with `id`
+```
+[ "childNodeId1", "childNodeId2", ... ]
+```
 
-`POST organizations/:org_id/surveys` - create a new survey
+`DELETE organizations/:org_id/containers/:id/children/:id2` - remove node `id2` from container with `id`
 
-`GET organizations/:org_id/surveys/:id` - return a specific survey
+### Surveys
 
-`PUT organizations/:org_id/surveys/:id` - update a specific survey
+`POST /organizations/:org_id/surveys` - create a new survey
 
-`PUT organizations/:org_id/surveys/:id/node/:id2` - attach survey `:id` to node `:id2`
+`GET /organizations/:org_id/surveys/:id` - return a specific survey
 
-`GET organizations/:org_id/surveys/node/:id` - return all attached surveys in tree starting at node `:id`
+`GET /organizations/:org_id/surveys?root={:id}` - get the group of surveys from all nodes on the tree below `:id`
 
+`PUT /organizations/:org_id/surveys/:id` - update a specific survey
 
 
-## Offerings
+## Registrations
 
 ---
 
-`GET /offerings` - super admin, retrieve all offerings
+`GET /organizations/:org_id/registrations/:id` - get a specific registration
 
-`GET /offerings/:region` - get offerings in a region
+`GET /organizations/:org_id/registrations?root=:id` - get all registrations beneath a specific node on a tree
 
-`GET /offerings/:id` - get a specific offering and all children (wraps /nodes/:id)
+`POST /organizations/:org_id/registrations` - attempt to register for offering `:id`
+```
+{ "offering": "OFFERING_ID", ... }
+```
 
-`PUT /offerings/:id/deoffer` - not implemented? (see above)
+`POST /organizations/:org_id/registrations` - batch of registrations from offline source, admin only, `content-type: text/csv`
 
-`GET /offerings/:id/survey` - get the group of surveys from all nodes on the tree below `:id`
+`PUT /organizations/:org_id/registrations/:id` - update limited set of information - likely only admin
 
-## Registrations
+`DELETE /organizations/:org_id/registrations/:id` - cancel a specific registration, optionally issues refunds
 
----
+## Payments
 
-`GET /registrations/:id` - get a specific registration
+---
 
-`GET /registrations/node/:id` - get all registrations beneath a specific node on a tree
+`GET /organizations/:org_id/payments/:id` - get a specific payment receipt
 
-`GET /registrations/user/:id` - get all registrations for a specific user
+`PUT /organizations/:org_id/payments/:id` - update a payment. If it's associated with a vendor (stripe), this should be restricted to issuing partial refunds only.
```

Comment:
we may need to rethink payments in terms of refunds actually...

Reply:
I think registrations and payments should be separated - registrations are on a per-occasion basis, so it's more "What specific occasions am I registered for/what is my class schedule?" whereas payments occur on the root node of an offering.

Reply:
I see... Yeah that makes sense. I sort of see it as which payments are associated with this registration, not that they have to be hard linked. Payments should definitely be their own thing.